ui/components/src/context/DashboardTimeZoneProvider.tsx

```diff
@@ -14,27 +14,44 @@
 import React, { createContext, ReactElement, useContext } from 'react';
 import { formatWithTimeZone, dateFormatOptionsWithTimeZone } from '../utils';
 
-export const TimeZoneContext = createContext<string | undefined>(undefined);
+type DashboardTimeZoneContextType = {
+  timeZone: string;
+  setTimeZone?: (timeZone: string) => void;
+};
+
+export const DashboardTimeZoneContext = createContext<DashboardTimeZoneContextType | undefined>(undefined);
 
-export interface TimeZoneProviderProps {
+export interface DashboardTimeZoneProviderProps {
   timeZone?: string;
+  setTimeZone?: (timeZone: string) => void;
   children?: React.ReactNode;
 }
 
-export function TimeZoneProvider(props: TimeZoneProviderProps): ReactElement {
-  const { children, timeZone } = props;
-  return <TimeZoneContext.Provider value={timeZone}>{children}</TimeZoneContext.Provider>;
+export function DashboardTimeZoneProvider(props: DashboardTimeZoneProviderProps): ReactElement {
+  const { timeZone = 'local', setTimeZone } = props;
+  return (
+    <DashboardTimeZoneContext.Provider value={{ timeZone, setTimeZone }}>
+      {props.children}
+    </DashboardTimeZoneContext.Provider>
+  );
 }
 
-export function useTimeZone(): {
+export function useDashboardTimeZone(): {
   timeZone: string;
+  setTimeZone: (timeZone: string) => void;
   formatWithUserTimeZone: (date: Date, formatString: string) => string;
   dateFormatOptionsWithUserTimeZone: (dateFormatOptions: Intl.DateTimeFormatOptions) => Intl.DateTimeFormatOptions;
 } {
-  const timeZone = useContext(TimeZoneContext);
+  const timeZoneContext = useContext(DashboardTimeZoneContext);
+  if (!timeZoneContext) {
```

Comment:
This would mean that a timezone provider is mandatory, this will be a breaking change for plugin embedding, maybe we don't need this provider at all and can place the timezone setting in the time range provider. We want to reduce the amount of providers needed and this can be a good first case as the timezone relates to the selected time and the time range is indeed mandatory from the dashboard perspective. WDYT?

Reply:
> place the timezone setting in the time range provider.

I would also prefer that, the time zone should be a property on the time range provider.

Reply:
For clarification, we meant to merge the timeZone context inside the timeRange context. And add a `timeZone` property(field) to the timeRange context. Not related to component props.